internal/core/application/fraud.go

```diff
@@ -44,10 +49,23 @@ func (s *service) reactToFraud(ctx context.Context, vtxo domain.Vtxo, mutx *sync
 
 	// If the vtxo wasn't settled we must broadcast a checkpoint tx.
 	if !vtxo.IsSettled() {
-		if err := s.broadcastCheckpointTx(ctx, vtxo); err != nil {
+		checkpointPsbt, err := s.broadcastCheckpointTx(ctx, vtxo)
+		if err != nil {
 			return fmt.Errorf("failed to broadcast checkpoint tx: %s", err)
 		}
 
+		ptx, err := psbt.NewFromRawBytes(strings.NewReader(checkpointPsbt), true)
+		if err != nil {
+			return err
+		}
+
+		go func() {
+			blockStamp := s.waitForConfirmation(context.Background(), ptx.UnsignedTx.TxID())
```

Comment:
can't we simply return height and time here and prevent adding `blockStamp` struct?

```suggestion
			blockHeight, blockTime := s.waitForConfirmation(context.Background(), ptx.UnsignedTx.TxID())
```